elements/src/components/pos-list/pos-list.tsx

```diff
@@ -1,32 +1,44 @@
-import { Thing } from '@pod-os/core';
+import { PodOS, Thing } from '@pod-os/core';
 import { Component, Element, Event, h, Prop, State } from '@stencil/core';
 import { ResourceAware, ResourceEventEmitter, subscribeResource } from '../events/ResourceAware';
+import { PodOsAware, PodOsEventEmitter, subscribePodOs } from '../events/PodOsAware';
+import { Subject } from 'rxjs';
 
 @Component({
   tag: 'pos-list',
   shadow: false,
 })
-export class PosList implements ResourceAware {
+export class PosList implements PodOsAware, ResourceAware {
   /**
    * URI of the predicate to follow
    */
   @Prop() rel: string;
+  /**
+   * URI of a class for which instances will be listed
+   */
+  @Prop() ifTypeof: string;
   /**
    * Whether listed resources should be fetched before being displayed
    */
   @Prop() fetch: boolean = false;
 
   @Element() host: HTMLElement;
   @State() error: string = null;
+  @State() os: PodOS;
   @State() resource: Thing;
   @State() items: string[] = [];
   @State() templateString: string;
 
   @Event({ eventName: 'pod-os:resource' })
   subscribeResource: ResourceEventEmitter;
+  @Event({ eventName: 'pod-os:init' })
+  subscribePodOs: PodOsEventEmitter;
+
+  private readonly disconnected$ = new Subject<void>();
 
   componentWillLoad() {
-    subscribeResource(this);
+    if (this.rel) subscribeResource(this);
```

Comment:
issue: There is an issue with both rel and if-type-of: The resource does not reflect changes to these attributes. The data has to be updated on each change of those. It's fine for me to approach this issue in a separate MR.